content/blogposts/2024/announcing-prompt-library.md

```diff
@@ -0,0 +1,149 @@
+
+- Java version: 1.6
+- Spring framework version: 3.x
+- Testing framework: JUnit 4"
+
+By sharing this prompt with your team, everyone can ensure that Cody generates code compatible with your legacy system.
+
+## 2. “Cody is not following my organization’s style guide”
+
+Suppose your organization has a specific Python coding style guide that mandates the use of snake_case for variable names and requires docstrings for all functions. However, Cody might generate code that doesn't adhere to these conventions, leading to inconsistencies in your codebase.
+
+```python
+def processData(inputData):
+    """Process the input data and return the result."""
+    processedData = inputData * 2  # This should be snake_case
+    return processedData
+```
+
+To solve this, you can create a comprehensive prompt that instructs Cody to follow your organization's style guide:
+
+"Generate Python code that adheres to our organization's style guide, which includes the following requirements:
+
+- Use snake_case for all variable and function names.
+- Include docstrings for all functions, describing their purpose and parameters.
+- Follow PEP 8 guidelines for code formatting, including indentation and line length."
+
+This prompt can be shared with your colleagues to ensure consistent code style across the team.
+
+## 3. “Cody is not generating tests in the right format”
+
+Imagine you are working with Golang and need Cody to generate unit tests that comply with your team’s conventions. Your organization follows a specific structure for tests, including the use of table-driven tests and `require` from the `testify` package for assertions.
+
+However, Cody might provide a simple test that does not align with your standards:
+
+```go
+func TestAdd(t *testing.T) {
+    result := Add(2, 3)
+    if result != 5 {
+        t.Errorf("expected 5, got %d", result)
+    }
+}
+
+```
+
+To ensure consistency, create a prompt in the Prompt Library specifying your test requirements:
+
+"Please generate Golang unit tests that adhere to the following conventions:
+
+- Use table-driven tests for different input cases.
+- Use the `require` package from `testify` for assertions.
+- Follow idiomatic Go test naming conventions and structure."
+
+When this prompt is used, Cody will generate tests like the following:
+
+```go
+func TestAdd(t *testing.T) {
+    testCases := []struct {
+        name     string
+        a, b     int
+        expected int
+    }{
+        {"add positive numbers", 2, 3, 5},
+        {"add zero", 0, 5, 5},
+        {"add negative numbers", -2, -3, -5},
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.name, func(t *testing.T) {
+            result := Add(tc.a, tc.b)
+            require.Equal(t, tc.expected, result)
+        })
+    }
+}
+
+```
+
+By saving and sharing this prompt within your team, you ensure that Cody consistently generates tests in the correct format, saving time and reducing review cycles.
+
```

Comment:
It would be good to add 2 more examples showing the different context that prompts can use. Such as selected code, or entire repository, to really showcase the capabilities and functionality of the prompts feature.

Reply:
It might also be interesting to talk about the different sharing capabilities, so some prompts you may want to set to private, others you may want to share within the org. This could go before the examples section imo.